can/interfaces/vector/canlib.py

```diff
@@ -586,15 +576,99 @@ def popup_vector_hw_configuration(wait_for_finish: int = 0) -> None:
         """
         xldriver.xlPopupHwConfig(ctypes.c_char_p(), ctypes.c_uint(wait_for_finish))
 
+    @staticmethod
+    def get_application_config(
+        app_name: str, app_channel: int, bus_type: xldefine.XL_BusTypes
+    ) -> Tuple[xldefine.XL_HardwareType, int, int]:
+        """Retrieve information for an application in Vector Hardware Configuration.
+
+        :param app_name:
+            The name of the application.
+        :param app_channel:
+            The channel of the application.
+        :param bus_type:
+            The bus type Enum e.g. `XL_BusTypes.XL_BUS_TYPE_CAN`
+        :return:
+            Returns a tuple of the hardware type, the hardware index and the
+            hardware channel.
+        :raises VectorError:
+            Raises a VectorError when the application name does not exist in
+            Vector Hardware Configuration.
+        """
+        hw_type = ctypes.c_uint()
+        hw_index = ctypes.c_uint()
+        hw_channel = ctypes.c_uint()
+
+        xldriver.xlGetApplConfig(
+            app_name.encode(),
+            app_channel,
+            hw_type,
+            hw_index,
+            hw_channel,
+            bus_type.value,
+        )
+        return xldefine.XL_HardwareType(hw_type.value), hw_index.value, hw_channel.value
+
+    @staticmethod
+    def set_application_config(
+        app_name: str,
+        app_channel: int,
+        hw_type: xldefine.XL_HardwareType,
+        hw_index: int,
+        hw_channel: int,
+        bus_type: xldefine.XL_BusTypes,
+    ) -> None:
+        """Modify the application settings in Vector Hardware Configuration.
+
+        :param app_name:
+            The name of the application. Creates a new application if it does
+            not exist yet.
+        :param app_channel:
+            The channel of the application.
+        :param hw_type:
+            The hardware type of the interface.
+            E.g XL_HardwareType.XL_HWTYPE_VIRTUAL
+        :param hw_index:
+            The index of the interface if multiple interface with the same
+            hardware type are present.
+        :param hw_channel:
+            The channel index of the interface.
+        :param bus_type:
+            The bus type of the interfaces, which should be
+            XL_BusTypes.XL_BUS_TYPE_CAN for most cases.
+        """
+        xldriver.xlSetApplConfig(
+            app_name.encode(),
+            app_channel,
+            hw_type.value,
+            hw_index,
+            hw_channel,
+            bus_type.value,
+        )
+
+    def set_timer_rate(self, timer_rate_ms: int) -> None:
```

Comment:
How would you use this method?

Reply:
You run `set_timer_rate(10)` once and the interface will generate an event with the tag `XL_TIMER` every 10ms. If you have an active notifier this will cause the method `handle_can_event()` or `handle_canfd_event` to be called.

It should be a relatively precise timer since it is hardware generated. But since i am in home office i did not really have an opportunity to test how reliable it is.

Reply:
XL_TIMER:
![XL_TIMER](https://user-images.githubusercontent.com/52598363/83687796-95a98080-a5ec-11ea-904d-f1cc930c3ee4.png)

Windows Multimedia Timer:
![MultimediaTimer](https://user-images.githubusercontent.com/52598363/83687821-a22dd900-a5ec-11ea-9edf-1c62b6caa312.png)

Tested with a full restbus simulation on virtual can with a VN1610 attached. The timing looks good but the XL_TIMER events just get dropped if another event occurs at the same time which is quite disappointing.

Even on my slow Notebook the Windows MM Timer works better.

So it really is only useful to tell if the bus is still alive.

Reply:
Update: it is actually quite decent, there was something else hogging my event loop.

![XL_TIMER](https://user-images.githubusercontent.com/52598363/83932204-58d2ba80-a7a1-11ea-88b2-51ccbf8f1deb.png)

Reply:
I think I read about it in the API documentation thinking it would be a good candidate for sending cycling messages, but if I remember correctly the timer was reset each time some other event occurred so it didn't feel very useful. Since we usually have a timeout when waiting for events to keep the application alive even when there are no messages I'm not sure what the use case is, but on the other hand it doesn't hurt to have more options.